src/lat/determinize-lattice-pruned.h

```diff
@@ -155,11 +155,14 @@ struct DeterminizeLatticePhonePrunedOptions {
   bool word_determinize;
   // minimize: if true, push and minimize after determinization.
   bool minimize;
+  // extra parameters for speed 
+  int max_loop;
+  int max_arcs;
   DeterminizeLatticePhonePrunedOptions(): delta(kDelta),
                                           max_mem(50000000),
                                           phone_determinize(true),
                                           word_determinize(true),
-                                          minimize(false) {}
+                                          minimize(false) , max_loop(-1), max_arcs(-1) {}
```

Comment:
these should be on separate lines to match the other options.

But I don't fully understand why you need to add these options, since  you're not registering them.